client/src/auth/index.jsx

```diff
@@ -1,20 +1,22 @@
 import React from 'react';
 import isAuth from './isAuth';
 import { Route, Redirect } from 'react-router-dom';
-import { checkPropTypes } from 'prop-types';
 
 const PrivateRoute = ({ component: Component, ...rest }) => {
   return (
     <Route
       {...rest}
-      render={props => {
+      component={props => {
```

Comment:
I think you should read a little more about the difference between `render` and `component`, this just made the warning disappear

[check this](https://reacttraining.com/react-router/web/api/Route/component)